app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -134,15 +134,14 @@ def update_usergroups(login)
                         .where(ExternalUsergroup.arel_table[:auth_source_id].eq(id))
                         .pluck(:id)
     # User's external user groups as of auth_source
-    usergroup_ids.concat(
-      ExternalUsergroup.where(auth_source_id: id)
-        .where(group_name_arel.in(external))
-        .pluck(:usergroup_id)
-    )
-
-    Usergroup.where(id: usergroup_ids.uniq).find_each do |usergroup|
-      refresh_usergroup_members(usergroup)
-    end
+    external_mapping_ids = ExternalUsergroup.where(auth_source_id: id)
+                                            .where(group_name_arel.in(external_names))
+                                            .pluck(:usergroup_id)
+
+    user.usergroup_ids -= (usergroup_ids - external_mapping_ids)
+    user.usergroup_ids += (external_mapping_ids - usergroup_ids)
+    # audit changes
+    user.save
```

Comment:
We should probably skip validations, just in case somehow the user is invalid:
```suggestion
    user.save(validate: false)
```